react/features/large-video/actions.js

```diff
@@ -20,6 +21,50 @@ import {
 
 declare var APP: Object;
 
+/**
+* Captures a screenshot of the video displayed on the large video.
+*
+* @returns {Function}
+*/
+export function captureLargeVideoScreenshot() {
+   return (dispatch: Dispatch<any>, getState: Function): Promise<Object> => {
+       const state = getState();
+       const largeVideo = state['features/large-video'];
+
+       if (!largeVideo) {
+           return Promise.resolve();
+       }
+       const tracks = state['features/base/tracks'];
+       const { jitsiTrack } = getTrackByMediaTypeAndParticipant(tracks, MEDIA_TYPE.VIDEO, largeVideo.participantId);
+       const videoStream = jitsiTrack.getOriginalStream();
+
+       // create a HTML canvas and draw video from the track on to the canvas.
+       const [ track ] = videoStream.getVideoTracks();
+       const { height, width } = track.getSettings() ?? track.getConstraints();
+       const canvas = document.createElement('canvas');
+       const ctx = canvas.getContext('2d');
+       const element = document.createElement('video');
+
+       element.height = parseInt(height, 10);
+       element.width = parseInt(width, 10);
+       element.autoplay = true;
+       element.srcObject = videoStream;
+       canvas.height = parseInt(height, 10);
+       canvas.width = parseInt(width, 10);
+
+       // wait for the video to load before drawing on to the canvas.
+       const promise = new Promise(resolve => {
+           element.onloadeddata = () => resolve();
+       });
+
+       return promise.then(() => {
+           ctx.drawImage(element, 0, 0, element.width, element.height);
+
+           return Promise.resolve(canvas.toDataURL('image/jpeg', 1.0));
```

Comment:
maybe destroy the video and canvas before returning?

Reply:
I added some cleanup code.